rust/sbp/src/de.rs

```diff
@@ -238,35 +292,34 @@ impl TimeoutSbpDecoder {
     }
 }
 
-impl Decoder for TimeoutSbpDecoder {
+impl dencode::Decoder for TimeoutSbpDecoder {
     type Item = Sbp;
     type Error = Error;
 
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         if self.last_msg_received.elapsed() > self.timeout_duration {
             return Err(std::io::Error::new(std::io::ErrorKind::TimedOut, "timeout").into());
         }
-        let res = self.decoder.decode(src);
-        if let Ok(Some(_)) = res {
+        let next = match FramerImpl.decode(src)? {
+            Some(frame) => frame.to_sbp().map(Some),
```

Comment:
map some.... ? got confused with all the naming not even sure if this is ok but it def doesn't look right, mapping everything to some

Reply:
that's pretty normal/idiomatic way to go from `Result<T>` to `Result<Option<T>>`. `Some`/enum variants are just [functions](https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/first-edition/enums.html#constructors-as-functions) so mapping them is normal

Reply:
that is super weird, thought we could bubble up from the frame then instead of doin the ? to unwrap the FramerImpl decode making us manaully cast it to some